mobile/test/blocs/camera_permission/camera_permission_bloc_test.dart

```diff
@@ -0,0 +1,454 @@
+import 'package:bloc_test/bloc_test.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mocktail/mocktail.dart';
+import 'package:openvine/blocs/camera_permission/camera_permission_bloc.dart';
+import 'package:permission_handler_platform_interface/permission_handler_platform_interface.dart';
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+
+class MockPermissionHandlerPlatform extends Mock
+    with MockPlatformInterfaceMixin
+    implements PermissionHandlerPlatform {}
+
+void main() {
+  group('CameraPermissionBloc', () {
+    late MockPermissionHandlerPlatform mockPlatform;
+
+    setUp(() {
+      mockPlatform = MockPermissionHandlerPlatform();
+      PermissionHandlerPlatform.instance = mockPlatform;
+    });
+
+    group('initial state', () {
+      test('is CameraPermissionInitial', () {
+        final bloc = CameraPermissionBloc();
+        expect(bloc.state, isA<CameraPermissionInitial>());
+      });
+    });
+
+    group('CameraPermissionRequest', () {
+      blocTest<CameraPermissionBloc, CameraPermissionState>(
+        'emits [Loaded(authorized)] when both permissions granted',
+        setUp: () {
+          // Set up initial state as canRequest
+          when(
+            () => mockPlatform.checkPermissionStatus(Permission.camera),
+          ).thenAnswer((_) async => PermissionStatus.denied);
+          when(
+            () => mockPlatform.checkPermissionStatus(Permission.microphone),
+          ).thenAnswer((_) async => PermissionStatus.denied);
+          // Set up request responses
+          when(
+            () => mockPlatform.requestPermissions([Permission.camera]),
+          ).thenAnswer(
+            (_) async => {Permission.camera: PermissionStatus.granted},
+          );
+          when(
+            () => mockPlatform.requestPermissions([Permission.microphone]),
+          ).thenAnswer(
+            (_) async => {Permission.microphone: PermissionStatus.granted},
+          );
+        },
+        build: () => CameraPermissionBloc(),
+        act: (bloc) async {
+          bloc.add(const CameraPermissionRefresh());
```

Comment:
Why CameraPermissionRefresh is needed here? I believe this is just testing CameraPermissionRequest which should be independent